### ISSUE TITLE
Windows stuck to invisible borders and minimizing all other windows

### ISSUE THREAD

**Issue by Palago** [digitalcreations/MaxTo#740]
Windows will not drag bast seemingly arbitrary points, sometimes it's another window's edge, and sometimes it's just random.  If you try to continue dragging it will minimize all other windows you have open

**To Reproduce**
On version 2.2.0 this happens almost any time I try to drag a window around.  I have reset to default settings and it still happens

**Expected behavior**
I can drag my windows around normally and without minimizing my other windows.  If I have snap to edges on I would expect a snap when the window passes by but then to be able to continue dragging

**Screenshots**
https://gyazo.com/0fca3d12131afb1d2c426e10735f92a9
Gif demonstration

**System information:**
Version 2004 (OS Build 19041.1052)
2.2.0 MaxTo Version

.json was not allowed as upload so renamed to a TXT
[config.txt](https://github.com/digitalcreations/MaxTo/files/6658076/config.txt)

**Edit**
Minimizing ended up being an issue with the windows "Shake to minimize" "feature" that I didn't know existed so thanks everyone for pointing that out

**Comment by rooby** [comment]
For me it seems to be the window snapping functionality. As you are dragging and the snap to edge is trying to reposition the window at the same time it causes the Windows 'shake to minimize all' functionality to trigger and everything minimises.
Seems to be more of an issue the larger the snap distance is.
Making snap distance 1px seems to maybe resolve it, but my preference is just to disable all the snap to settings.

The snap to functionality also causes some other strange behaviour, like when you drag past something and it snaps then sometimes the position your mouse cursor was in relation to the window you were moving changes, which can create a pretty unusual user experience sometimes.

I don't think I ever noticed these issues before version 2.2.0.

**Comment by eikowagenknecht** [comment]
Happens to me as well since upgrading to 2.2.0. I have windows "shake to minimize" disabled, so it only snaps windows to arbitrary positions and doesn't allow me to move them any further. This is really frustrating and makes MaxTo unusable.

**Comment by KimmoJ** [comment]
Same. Extremely annoying. When I pull windows past lines where I have maxto regions set, they glue themselves to those lines, slide up and down along those lines, other windows minimize, it's a damned mess. 

I'm pretty sure I turned off auto-updating. Again. I may have turned it on at some point after being lulled into a false sense of security I guess - I won't make that mistake again I guess. Finally managed to install 2.1.5 after downloading it, but I had to disable my network or else it just instantly sucked down 2.2 again and re-broke the system.

**Comment by therealptrk** [comment]
Same here since last Update. Extremly annoying! :/

**Comment by Palago** [comment]
Minimizing was the shaking option, didn't know that existed so thanks for pointing that out, the other issues still persist.

**Comment by vegardlarsen** [comment]
Can you try downgrading to version 2.2.0-beta.2 ([available from here](https://maxto.net/release/beta)), and see if the same issue is present there? That will help me figure out where we broke this.

**Comment by Palago** [comment]
> 
> 
> Can you try downgrading to version 2.2.0-beta.2 ([available from here](https://maxto.net/release/beta)), and see if the same issue is present there? That will help me figure out where we broke this.

Still Happening on this version, no noticeable difference

**Comment by vegardlarsen** [comment]
@Palago Can you go back one step further, to 2.2.0-beta.1?

**Comment by Palago** [comment]
@vegardlarsen Sorry apparently I'd installed it incorrectly before, just wiped the AppData/MaxTo and retried and beta.2 seems to fix it

**Comment by Nolsoee** [comment]
I had the same issue. Really annoying. I fixed it by going into MaxTo settings. Then window bahavior. Then under window snapping you can turn it of or on. I ended up with keeping it turned on but changed the snap distance to 3 pixels instead of 10. For me this fixed the issue

**Comment by lucaslink** [comment]
With all 3 snapping options checked on and pixel distance at 10, I experience similar behavior to whats mentioned above, with an additional caveat that the window gets stuck at the snapped position and can no longer be moved left or right, but only expanded horizontally. 
@Nolsoee suggestion of 3px distance with only Window Edges and Monitor Edges selected returns MaxTo to expected behavior.

**Comment by mclindgren** [comment]
Hi,

just wanted to write a comment saying I also noticed the same issues starting earlier this week. 
I use MaxTo because I use an ultrawide monitor 3440*1440 and it makes it easier to divide the screen into three sections.

I've not had any issues before it updated to 2.2.0.

Windows 10 Home
Version	21H1
OS-version	19043.1052

***Update***
I noticed that when a window gets stuck, if I maximize window and then put it back to normal I can move it again until it get stuck to the invisible borders again. Also, when a window gets stuck, and I try to drag it, it has the same effect as if I take a window and "shake it", It triggers the minimize feature for all other open windows.
